src/AuthModule/Components/Register/Register.tsx

```diff
@@ -1,7 +1,432 @@
+import { useForm } from "react-hook-form";
+import { toast } from "react-toastify";
+import { baseUrl } from "../../../Constants/Components/Urls";
+import {
+  countryValidation,
+  emailValidation,
+  passwordValidation,
+  phoneNumberValidation,
+  userNameValidation,
+} from "../Validators/Validators";
+import axios from "axios";
+import avatar from '../../../assets/avatar2.jpg'
 
+
+
+
+const VisuallyHiddenInput = styled('input')({
+  clip: 'rect(0 0 0 0)',
+  clipPath: 'inset(50%)',
+  height: 1,
+  overflow: 'hidden',
+  position: 'absolute',
+  bottom: 0,
+  left: 0,
+  whiteSpace: 'nowrap',
+  width: 1,
+});
 export default function Register() {
+  // const defaultTheme = createTheme();
+
+  // const [isLoading, setIsLoading] = useState<boolean>(false);
+  const navigate = useNavigate();
+  const {
+    register,
+    handleSubmit,
+    formState: { errors },
+    watch,
+  } = useForm();
+  const confirmPassword = watch("password");
+
+  const [userImage,setUserImage]=useState(avatar);
+
+  type RegisterData = {
+    userName: string;
+    phoneNumber: string;
+    country: string;
+    email: string;
+    password: string;
+    confirmPassword: string;
+    role:string;
+    profileImage:string;
+  };
+
+  const appendRegisterFormData=(data:RegisterData)=>{
+    let formData=new FormData();
+    formData.append("userName",data.userName);
+    formData.append("phoneNumber",data.phoneNumber);
+    formData.append("country",data.country);
+    formData.append("email",data.email);
+    formData.append("password",data.password);
+    formData.append("confirmPassword",data.confirmPassword);
+    formData.append("role",data.role);
+    formData.append("profileImage",data.profileImage[0]);
+    return formData;
+  }
+  async function handleRegister(data: RegisterData) {
+    console.log(data);
+    let registerDataForm= appendRegisterFormData(data);
+
+    try {
+      const response = await axios.post(`${baseUrl}/admin/users`, registerDataForm);
+      // console.log(response);
+      toast.success('Account Created Succefully');
+      navigate('/Authentication');
+    } catch (error) {
+      // console.log(error?.response?.data?.message);
+      toast.error(error?.response?.data?.message||'Somthing went wrong!');
+    }
+  }
   return (
-    <div>Register</div>
-  )
+    <>
+      {/* <ThemeProvider theme={defaultTheme}> */}
+      <Grid container component="main" sx={{ height: "100vh" }}>
+        {/* <CssBaseline /> */}
+
+        <Grid
+          item
+          xs={12}
+          sm={12}
+          md={5}
+          lg={6}
+          component={Paper}
+          elevation={6}
+          square
+          sx={{ height: "100vh" }}
+          style={{ overflow: "auto" }}
+          className="pageOverflow"
+        >
+          <Box
+            sx={{
+              my: 4,
+              mx: 4,
+            }}
+          >
+            <Link
+              to={"/"}
+              style={{
+                color: "#3252df",
```

Comment:
We shouldn't be using hardcoded colors, as it will make it difficult to update later or add new features such as dark/light mode.

To use MUI's pallet, we have `<Typography component={<Link .../>}`/> and access the theme using the `sx` prop.

Reply:
It didn't work

Reply:
It should be working as it can take a react component as mentioned below:
![image](https://github.com/SalmaAboArab/Hotel-Management-System/assets/64327685/674562ad-1cd9-4aed-a254-f5346d1e993f)

what did you try here?

Reply:
I tryed to use typography and put component with <Link> as you told me but it gave an error

Reply:
Yeah it seems that it takes only a component with creating it such as  `</Typography component={Link}/>`
You can do this instead:
```jsx
 
              <Typography
                variant="h5"
                sx={{
                  color: "primary.main",
                  fontWeight: 600,
                  "& span": { color: "black" },
                }}
              >
                <Link
                  to={"/"}
                  style={{
                    textDecoration: "none",
                  }}
                >
                  Stay
                  <span>cation</span>
                </Link>
              </Typography>

```

Reply:
I tried it and the color didn't affect word Stay
It only affects the line under the word

Reply:
I checked the color and it is working properly or do yo mean something else?
![image](https://github.com/SalmaAboArab/Hotel-Management-System/assets/64327685/47ca2646-1b50-465c-97d6-620ac8b7f113)

Reply:
![image](https://github.com/SalmaAboArab/Hotel-Management-System/assets/116819016/e3b4c099-d0db-4c2e-9c7b-f30363a233fd)
the default color for the project is blue 
but primary.main that we used in the code is another degree of blue
in this picture the primary.main is only affecting the under line of the word

Reply:
![image](https://github.com/SalmaAboArab/Hotel-Management-System/assets/116819016/975708b0-6957-4df8-a1e2-687f0eda1b1b)
this is my code
just like yours